server/src/main/java/org/opensearch/LegacyESVersion.java

```diff
@@ -156,4 +216,75 @@ public boolean isBeta() {
     public boolean isAlpha() {
         return major < 5 ? false :  build < 25;
     }
+
+    /**
+     * Returns the version given its string representation, current version if the argument is null or empty
+     */
+    public static Version fromString(String version) {
```

Comment:
Seems like this inherited from Version.java, duplicated?

Reply:
No.. it's a static method that needs to call the `LegacyESVersion.fromStringSlow` so it needs to be overloaded here. If inherited it will call the wrong implementation. (This is party of the hacky-ness that needs to be reworked over time).